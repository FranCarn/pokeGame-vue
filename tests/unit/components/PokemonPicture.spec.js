import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture.vue";

describe("pokemonPicture tests", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });
    jest.clearAllMocks();
  });
  afterEach(() => {
    wrapper = null;
  });
  test("should match with snapshot", () => {
    expect(wrapper).toMatchSnapshot("pokemonPicture");
  });
  test("should show hidden picture with pokemon 100", () => {
    const [hideImg, showImg] = wrapper.findAll("img");
    expect(hideImg.exists()).toBeTruthy();
    expect(hideImg.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
    expect(showImg).toBeUndefined();
    expect(hideImg.classes("hidden-pokemon")).toBeTruthy();
  });
  test("should show pokemon image if showpicture:true in props", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });
    const img = wrapper.find("img");

    expect(img.exists()).toBeTruthy();
    expect(img.classes("fade-in")).toBeTruthy();
  });
});
