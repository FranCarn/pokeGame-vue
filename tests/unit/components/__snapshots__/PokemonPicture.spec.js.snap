// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonPicture tests should match with snapshot: pokemonPicture 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="pokemon-container"
      >
        <img
          alt="pokemon"
          class="hidden-pokemon"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
        />
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
          "pokemonId": 100,
          "showPokemon": false,
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
            "pokemonId": 100,
            "showPokemon": false,
          },
          "_value": Object {
            "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
            "pokemonId": 100,
            "showPokemon": false,
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 5,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
          "pokemonId": 100,
          "showPokemon": false,
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
          "pokemonId": 100,
          "showPokemon": false,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "_depsLength": 5,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
          "pokemonId": 100,
          "showPokemon": false,
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "constructor": 0,
            "imgSrc": 4,
            "isVueInstance": 0,
            "pokemonId": 3,
            "showPokemon": 3,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
            "pokemonId": 100,
            "showPokemon": false,
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 2,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "pokemonId": 100,
            "showPokemon": false,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "pokemonId": Object {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "showPokemon": Object {
                "0": true,
                "1": true,
                "default": false,
                "required": true,
                "type": [Function],
              },
            },
            Array [
              "showPokemon",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
            "pokemonId": 100,
            "showPokemon": false,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "_depsLength": 1,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_setter": [Function],
                  "_value": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
                  "dep": Map {
                    ReactiveEffect {
                      "_depsLength": 2,
                      "_dirtyLevel": 0,
                      "_runnings": 0,
                      "_shouldSchedule": false,
                      "_trackId": 1,
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Map {
                          [Circular] => 1,
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "scheduler": [Function],
                      "trigger": [Function],
                    } => 1,
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "scheduler": [Function],
                "trigger": [Function],
              },
              ReactiveEffect {
                "_depsLength": 2,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": Array [
                  "src",
                ],
                "el": <img
                  alt="pokemon"
                  class="hidden-pokemon"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
                />,
                "key": 0,
                "patchFlag": 8,
                "props": Object {
                  "alt": "pokemon",
                  "class": "hidden-pokemon",
                  "key": 0,
                  "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": Array [
                  "src",
                ],
                "el": <img
                  alt="pokemon"
                  class="hidden-pokemon"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
                />,
                "key": 0,
                "patchFlag": 8,
                "props": Object {
                  "alt": "pokemon",
                  "class": "hidden-pokemon",
                  "key": 0,
                  "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="pokemon-container"
            >
              <img
                alt="pokemon"
                class="hidden-pokemon"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
              />
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "pokemon-container",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "computed": Object {
              "imgSrc": [Function],
            },
            "props": Object {
              "pokemonId": Object {
                "required": true,
                "type": [Function],
              },
              "showPokemon": Object {
                "default": false,
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="pokemon-container"
        >
          <img
            alt="pokemon"
            class="hidden-pokemon"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "pokemonId": 100,
          "ref": "VTU_COMPONENT",
          "showPokemon": false,
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "computed": Object {
            "imgSrc": [Function],
          },
          "props": Object {
            "pokemonId": Object {
              "required": true,
              "type": [Function],
            },
            "showPokemon": Object {
              "default": false,
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="pokemon-container"
        >
          <img
            alt="pokemon"
            class="hidden-pokemon"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.15",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
    "pokemonId": 100,
    "showPokemon": false,
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "imgSrc": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
      "pokemonId": 100,
      "showPokemon": false,
    },
  },
  "wrapperElement": <div
    class="pokemon-container"
  >
    <img
      alt="pokemon"
      class="hidden-pokemon"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    />
  </div>,
}
`;
